### ISSUE TITLE
remove any uneccessary login.test from yaml indexers

### ISSUE THREAD

**Issue by garfield69** [Jackett/Jackett#7838]
For those Cardigann indexers that have good enough login.error trapping in the login block, remove the subsequent login.test (usually a check for a logout href on the main page).
this test only adds to the number of requests made to the site, and if the error trap catches the login failure, then the subsequent test is not really necessary.
Some sites have poor error trapping, and the test is the only way to check the login was successful, so each indexer will need to be tested carefully.
- [ ] [ ] 0daykiev.yml
- [ ] [ ] 1337x.yml
- [ ] [ ] 2fast4you.yml
- [ ] [ ] 3dtorrents.yml
- [ ] [ ] 3evils.yml
- [ ] [ ] 4thd.yml
- [ ] [ ] 52pt.yml
- [ ] [ ] 7torrents.yml
- [ ] [ ] academictorrents.yml
- [ ] [ ] acgrip.yml
- [ ] [ ] acgsou.yml
- [ ] [ ] acidlounge.yml
- [ ] [ ] aftershock.yml
- [ ] [ ] alexfilm.yml
- [ ] [ ] amigosshare.yml
- [ ] [ ] anirena.yml
- [ ] [ ] arabafenice.yml
- [ ] [ ] arenabg.yml
- [ ] [ ] asgaard.yml
- [ ] [ ] asiancinema.yml
- [ ] [ ] ast4u.yml
- [ ] [ ] audiobookbay.yml
- [ ] [ ] audiobooktorrents.yml
- [ ] [ ] audionews.yml
- [ ] [ ] awesomehd.yml
- [ ] [ ] backups.yml
- [ ] [ ] badasstorrents.yml
- [ ] [ ] baibako.yml
- [ ] [ ] beitai.yml
- [ ] [ ] beyond-hd-oneurl.yml
- [ ] [ ] beyond-hd.yml
- [ ] [ ] bibliotik.yml
- [ ] [ ] bigfangroup.yml
- [ ] [ ] bigtorrent.yml
- [ ] [ ] bigtower.yml
- [ ] [ ] bit-titan.yml
- [ ] [ ] bithorlo.yml
- [ ] [ ] bithumen.yml
- [ ] [ ] bitru.yml
- [ ] [ ] bitspyder.yml
- [ ] [ ] bittorrentam.yml
- [ ] [ ] Bittorrentfiles.yml
- [ ] [ ] bitturk.yml
- [ ] [ ] bluebird.yml
- [ ] [ ] blutopia.yml
- [ ] [ ] booktracker.yml
- [ ] [ ] boxingtorrents.yml
- [ ] [ ] brasiltracker.yml
- [ ] [ ] broadcity.yml
- [ ] [ ] brobits.yml
- [ ] [ ] btdb.yml
- [ ] [ ] btdigg.yml
- [ ] [ ] btetree.yml
- [ ] [ ] bteye.yml
- [ ] [ ] btgigs.yml
- [ ] [ ] btnext.yml
- [ ] [ ] btschool.yml
- [ ] [ ] btsow.yml
- [ ] [ ] carpathians.yml
- [ ] [ ] cartoonchaos.yml
- [ ] [ ] casatorrent.yml
- [ ] [ ] casstudiotv.yml
- [ ] [ ] ccfbits.yml
- [ ] [ ] channelx.yml
- [ ] [ ] chdbits.yml
- [ ] [ ] cili180.yml
- [ ] [ ] cinemageddon.yml
- [ ] [ ] cinemamovies.yml
- [ ] [ ] cinematik.yml
- [ ] [ ] classix.yml
- [ ] [ ] concen.yml
- [ ] [ ] concertos.yml
- [ ] [ ] cpabien.yml
- [ ] [ ] cpasbienclone.yml
- [ ] [ ] crazyhd.yml
- [ ] [ ] crazyscorner.yml
- [ ] [ ] crazyspirits.yml
- [ ] [ ] crnaberza.yml
- [ ] [ ] cztorrent.yml
- [ ] [ ] DasUnerwartete.yml
- [ ] [ ] datascene.yml
- [ ] [ ] deildu.yml
- [ ] [ ] demonoid.yml
- [ ] [ ] demonsite.yml
- [ ] [ ] desireleasers.yml
- [ ] [ ] desitorrents.yml
- [ ] [ ] diablotorrent.yml
- [ ] [ ] divteam.yml
- [ ] [ ] dmhy.yml
- [ ] [ ] documentarytorrents.yml
- [ ] [ ] downloadville.yml
- [ ] [ ] dragonworldreloaded.yml
- [ ] [ ] dxdhd.yml
- [ ] [ ] dxp.yml
- [ ] [ ] ebookparadijs.yml
- [ ] [ ] ebooks-shares.yml
- [ ] [ ] efectodoppler.yml
- [ ] [ ] elitetorrent-biz.yml
- [ ] [ ] empornium.yml
- [ ] [ ] eniahd.yml
- [ ] [ ] erzsebet.yml
- [ ] [ ] esharenet.yml
- [ ] [ ] estone.yml
- [ ] [ ] estrenosdtl.yml
- [ ] [ ] ethor.yml
- [ ] [ ] ettv.yml
- [ ] [ ] exkinoray.yml
- [ ] [ ] extratorrent-cd.yml
- [ ] [ ] extremetorrents.yml
- [ ] [ ] extremlymtorrents.yml
- [ ] [ ] exttorrents.yml
- [ ] [ ] eztv.yml
- [ ] [ ] fanoin.yml
- [ ] [ ] fantasticheaven.yml
- [ ] [ ] feedurneed.yml
- [ ] [ ] femdomcult.yml
- [ ] [ ] filebase.yml
- [ ] [ ] film-paleis.yml
- [ ] [ ] filmsclub.yml
- [ ] [ ] finvip.yml
- [ ] [ ] firebit.yml
- [ ] [ ] focusx.yml
- [ ] [ ] freetorrent.yml
- [ ] [ ] frozenlayer.yml
- [ ] [ ] fullmixmusic.yml
- [ ] [ ] funkytorrents.yml
- [ ] [ ] galeriens.yml
- [ ] [ ] gamestorrents.yml
- [ ] [ ] gay-torrents.yml
- [ ] [ ] gay-torrentsorg.yml
- [ ] [ ] gaytorrentru.yml
- [ ] [ ] generationfree.yml
- [ ] [ ] gfxpeers.yml
- [ ] [ ] gigatorrents.yml
- [ ] [ ] girotorrent.yml
- [ ] [ ] gktorrent.yml
- [ ] [ ] glodls.yml
- [ ] [ ] greekteam.yml
- [ ] [ ] hachede.yml
- [ ] [ ] hamsterstudio.yml
- [ ] [ ] hd4fans.yml
- [ ] [ ] hdarea.yml
- [ ] [ ] hdbits.yml
- [ ] [ ] hdbitscom.yml
- [ ] [ ] hdcenter.yml
- [ ] [ ] hdchina.yml
- [ ] [ ] hdcity.yml
- [ ] [ ] hddisk.yml
- [ ] [ ] hddolby.yml
- [ ] [ ] hdforever.yml
- [ ] [ ] hdhome.yml
- [ ] [ ] hdme.yml
- [ ] [ ] hdreactor.yml
- [ ] [ ] hdroute.yml
- [ ] [ ] hdsky.yml
- [ ] [ ] hdspain.yml
- [ ] [ ] hdtime.yml
- [ ] [ ] hdtorrentsit.yml
- [ ] [ ] hdturk.yml
- [ ] [ ] hdu.yml
- [ ] [ ] hdzone.yml
- [ ] [ ] hon3yhd.yml
- [ ] [ ] horrorsite.yml
- [ ] [ ] hqsource.yml
- [ ] [ ] hush.yml
- [ ] [ ] ibit.yml
- [ ] [ ] icetorrent.yml
- [ ] [ ] idope.yml
- [ ] [ ] ilcorsaroblu.yml
- [ ] [ ] ilcorsaronero.yml
- [ ] [ ] immortuos.yml
- [ ] [ ] inperil.yml
- [ ] [ ] insanetracker.yml
- [ ] [ ] isohunt2.yml
- [ ] [ ] itorrent.yml
- [ ] [ ] iv-torrents.yml
- [ ] [ ] jpopsuki.yml
- [ ] [ ] kapaki.yml
- [ ] [ ] karagarga.yml
- [ ] [ ] kickasstorrent-kathow.yml
- [ ] [ ] kickasstorrent.yml
- [ ] [ ] kinonavse100.yml
- [ ] [ ] kinozal.yml
- [ ] [ ] korsar.yml
- [ ] [ ] lastfiles.yml
- [ ] [ ] leaguehd.yml
- [ ] [ ] learnflakes.yml
- [ ] [ ] lechaudron.yml
- [ ] [ ] legacyhd.yml
- [ ] [ ] legittorrents.yml
- [ ] [ ] leporno.yml
- [ ] [ ] lesaloon.yml
- [ ] [ ] libranet.yml
- [ ] [ ] limetorrents.yml
- [ ] [ ] limitofeden.yml
- [ ] [ ] linkomanija.yml
- [ ] [ ] linuxtracker.yml
- [ ] [ ] list.txt
- [ ] [ ] losslessclub.yml
- [ ] [ ] mactorrents.yml
- [ ] [ ] magico.yml
- [ ] [ ] magnet4you.yml
- [ ] [ ] magnetdl.yml
- [ ] [ ] majomparade.yml
- [ ] [ ] marinetracker.yml
- [ ] [ ] metal-iplay-ro.yml
- [ ] [ ] metaltracker.yml
- [ ] [ ] microbit.yml
- [ ] [ ] mma-torrents.yml
- [ ] [ ] moecat.yml
- [ ] [ ] mononokebt.yml
- [ ] [ ] monova.yml
- [ ] [ ] movcr.yml
- [ ] [ ] moviesdvdr.yml
- [ ] [ ] mteamtp.yml
- [ ] [ ] mteamtp2fa.yml
- [ ] [ ] muziekfabriek.yml
- [ ] [ ] mvgroupforum.yml
- [ ] [ ] mvgroupmain.yml
- [ ] [ ] mypornclub.yml
- [ ] [ ] myspleen.yml
- [ ] [ ] nbtorrents.yml
- [ ] [ ] netcosmo.yml
- [ ] [ ] nethd.yml
- [ ] [ ] netlab.yml
- [ ] [ ] newretro.yml
- [ ] [ ] newstudio.yml
- [ ] [ ] newstudiol.yml
- [ ] [ ] nitro.yml
- [ ] [ ] nnm-club.yml
- [ ] [ ] nntt.yml
- [ ] [ ] nostalgic.yml
- [ ] [ ] nyaa-pantsu.yml
- [ ] [ ] nyaasi.yml
- [ ] [ ] onejav.yml
- [ ] [ ] onlineselfeducation.yml
- [ ] [ ] ourbits.yml
- [ ] [ ] oxtorrent.yml
- [ ] [ ] p2pbg.yml
- [ ] [ ] p2pelite.yml
- [ ] [ ] parnuxi.yml
- [ ] [ ] peersfm.yml
- [ ] [ ] piratbit.yml
- [ ] [ ] piratbitl.yml
- [ ] [ ] pirateiro.yml
- [ ] [ ] pixelcove.yml
- [ ] [ ] pleasuredome.yml
- [ ] [ ] polishsource.yml
- [ ] [ ] pornbay.yml
- [ ] [ ] pornbits.yml
- [ ] [ ] pornleech.yml
- [ ] [ ] pornolive.yml
- [ ] [ ] pornorip.yml
- [ ] [ ] pornotor.yml
- [ ] [ ] pornotorrent.yml
- [ ] [ ] proaudiotorrents.yml
- [ ] [ ] prostylex.yml
- [ ] [ ] pt99.yml
- [ ] [ ] ptfiles.yml
- [ ] [ ] pthome.yml
- [ ] [ ] puntotorrent.yml
- [ ] [ ] pwtorrents.yml
- [ ] [ ] r3vwtf.yml
- [ ] [ ] racing4everyone.yml
- [ ] [ ] racingforme.yml
- [ ] [ ] rapidzona.yml
- [ ] [ ] redstartorrent.yml
- [ ] [ ] retroflix.yml
- [ ] [ ] riperam.yml
- [ ] [ ] ripstracker.yml
- [ ] [ ] rns.yml
- [ ] [ ] rockbox.yml
- [ ] [ ] rodvd.yml
- [ ] [ ] rptorrents.yml
- [ ] [ ] rus-media.yml
- [ ] [ ] rustorka.yml
- [ ] [ ] rutor.yml
- [ ] [ ] rutracker-ru.yml
- [ ] [ ] scenefz.yml
- [ ] [ ] scenepalace.yml
- [ ] [ ] scenerush.yml
- [ ] [ ] sdbits.yml
- [ ] [ ] seedfile.yml
- [ ] [ ] shareisland.yml
- [ ] [ ] sharespacedb.yml
- [ ] [ ] shareuniversity.yml
- [ ] [ ] sharewood.yml
- [ ] [ ] shellife.yml
- [ ] [ ] shokweb.yml
- [ ] [ ] siambit.yml
- [ ] [ ] sktorrent.yml
- [ ] [ ] skytorrentsclone.yml
- [ ] [ ] snowpt.yml
- [ ] [ ] sosulki.yml
- [ ] [ ] soundpark.yml
- [ ] [ ] spacetorrent.yml
- [ ] [ ] speedtorrentreloaded.yml
- [ ] [ ] spiritofrevolution.yml
- [ ] [ ] sporthd.yml
- [ ] [ ] sportscult.yml
- [ ] [ ] springsunday.yml
- [ ] [ ] sukebei-pantsu.yml
- [ ] [ ] sukebeinyaasi.yml
- [ ] [ ] takeabyte.yml
- [ ] [ ] tapochek.yml
- [ ] [ ] tasmanit.yml
- [ ] [ ] teamhd.yml
- [ ] [ ] teamos.yml
- [ ] [ ] tekno3d.yml
- [ ] [ ] tellytorrent.yml
- [ ] [ ] tenyardtracker.yml
- [ ] [ ] the-madhouse.yml
- [ ] [ ] theaudioscene.yml
- [ ] [ ] theempire.yml
- [ ] [ ] thefallingangels.yml
- [ ] [ ] thegeeks.yml
- [ ] [ ] thehorrorcharnel.yml
- [ ] [ ] theoccult.yml
- [ ] [ ] thepiratebay.yml
- [ ] [ ] theplace.yml
- [ ] [ ] theshinning.yml
- [ ] [ ] theshow.yml
- [ ] [ ] thevault.yml
- [ ] [ ] tlfbits.yml
- [ ] [ ] tokyotosho.yml
- [ ] [ ] topnow.yml
- [ ] [ ] torlock.yml
- [ ] [ ] toros.yml
- [ ] [ ] torrent-explosiv.yml
- [ ] [ ] torrent-pirat.yml
- [ ] [ ] torrent-turk.yml
- [ ] [ ] torrent4you.yml
- [ ] [ ] torrent9.yml
- [ ] [ ] torrent9clone.yml
- [ ] [ ] torrentbd.yml
- [ ] [ ] torrentccf.yml
- [ ] [ ] torrentdownload.yml
- [ ] [ ] torrentdownloads.yml
- [ ] [ ] torrentfactory.yml
- [ ] [ ] torrentfunk.yml
- [ ] [ ] torrentgalaxyorg.yml
- [ ] [ ] torrenthr.yml
- [ ] [ ] torrenting.yml
- [ ] [ ] torrentkitty.yml
- [ ] [ ] torrentland.yml
- [ ] [ ] torrentleech-pl.yml
- [ ] [ ] torrentlt.yml
- [ ] [ ] torrentoyunindir.yml
- [ ] [ ] torrentparadise.yml
- [ ] [ ] torrentproject2.yml
- [ ] [ ] torrentquest.yml
- [ ] [ ] torrentrex.yml
- [ ] [ ] torrentsectorcrew.yml
- [ ] [ ] torrentslocal.yml
- [ ] [ ] torrentv.yml
- [ ] [ ] torrentview.yml
- [ ] [ ] torrentz2.yml
- [ ] [ ] totallykids.yml
- [ ] [ ] totheglory.yml
- [ ] [ ] trancetraffic.yml
- [ ] [ ] trezzor.yml
- [ ] [ ] tribalmixes.yml
- [ ] [ ] ttsweb.yml
- [ ] [ ] turktorrent.yml
- [ ] [ ] tvchaosuk.yml
- [ ] [ ] twilight.yml
- [ ] [ ] twilightszoom.yml
- [ ] [ ] u-torrents.yml
- [ ] [ ] u2.yml
- [ ] [ ] uhdbits.yml
- [ ] [ ] underverse.yml
- [ ] [ ] underversel.yml
- [ ] [ ] uniondht.yml
- [ ] [ ] unionfansub.yml
- [ ] [ ] uniongang.yml
- [ ] [ ] unlimitz.yml
- [ ] [ ] vanila.yml
- [ ] [ ] vizuk.yml
- [ ] [ ] wdt.yml
- [ ] [ ] wihd.yml
- [ ] [ ] world-of-tomorrow.yml
- [ ] [ ] x-ite.me.yml
- [ ] [ ] xbytes2.yml
- [ ] [ ] xtremefile.yml
- [ ] [ ] xtremezone.yml
- [ ] [ ] xwtclassics.yml
- [ ] [ ] xwtorrents.yml
- [ ] [ ] xxxtor.yml
- [ ] [ ] yggcookie.yml
- [ ] [ ] yggtorrent.yml
- [ ] [ ] yourbittorrent.yml
- [ ] [ ] yourexotic.yml
- [ ] [ ] zamundanet.yml
- [ ] [ ] zelkaorg.yml
- [ ] [ ] zooqle.yml
- [ ] [ ] ztracker.yml
first box is for test completed, second box for relevant commit id.

**Comment by ngosang** [comment]
TLDR; Don't apply this in all trackers, just in UNIT3D.

The Cardigann parser works as follows when you make a search:
1. Send the request with the parameters and cookie.
2. Check if the response is a redirect => if YES do relogin and perform the search again [code](91812d94b9aca29e97db99c3ef13d44e709ce71e)
3. Check if the response contains the `login.test.selector` => if NO do relogin and perform the search again [code](https://github.com/Jackett/Jackett/blob/91812d94b9aca29e97db99c3ef13d44e709ce71e/src/Jackett.Common/Indexers/CardigannIndexer.cs#L837)
4. Check if the response HTTP code is 401 Unauthorized=> if YES do relogin and perform the search again [code](https://github.com/Jackett/Jackett/blob/91812d94b9aca29e97db99c3ef13d44e709ce71e/src/Jackett.Common/Indexers/CardigannIndexer.cs#L442)
5. Check if the response contains the `login.error.selector`=> if YES do relogin and perform the search again [code](https://github.com/Jackett/Jackett/blob/91812d94b9aca29e97db99c3ef13d44e709ce71e/src/Jackett.Common/Indexers/CardigannIndexer.cs#L454)

I think we should include the `login.test.selector` and `login.error.selector` in all trackers. This is a double check that can detect problems in the trackers. In a normal scenario these checks don't make more requests, just one for perform the search, that's all.
The exception to this is UNIT3D because the torrent search URL doesn't return the complete web page, only the table with the torrents. In the `login.test.selector` we were looking for the logout link that is not in the search response. Instead of removing the `login.test.selector`, we could try to find the table with torrents to keep both checks.

Related #7834

**Comment by garfield69** [comment]
ok, i'll close this one then. I havent started, I went for a mountain bike ride for some fresh air and i've just come back ready to get stuck in, so your time is perfect.

re: unit3d, HDVinnie the author of unit3d gave us the existing search path to use deliberately because using the one that the web site uses was causing issues some time in the unit3d past.
